### PR TITLE
Add wildcard filtering support when dump on exceptions

### PR DESCRIPTION
**Pull request by oldzhu**
Add wildcard filtering support when dump on exceptions #153

### CODE REVIEW

File: profiler/inc/profilerstring.h [hunk 1]
```diff
@@ -333,3 +357,15 @@ inline bool EndsWith(const String &lhs, const String &rhs)
 
     return true;
 }
+
+inline WCHAR *getWCHARs(const String &src)
+{
+    size_t len = src.Length();
+    WCHAR* tempbuff = new WCHAR[len+1];
```
Comment (MarioHewardt) [inline profiler/inc/profilerstring.h:364]: Check allocation result

File: profiler/src/ProcDumpProfiler.cpp [hunk 2]
```diff
@@ -11,6 +11,143 @@
 INITIALIZE_EASYLOGGINGPP
 
 
+//--------------------------------------------------------------------
+//
+// WildcardSearch - Search string supports '*' anywhere and any number of times
+//
+//--------------------------------------------------------------------
+bool WildcardSearch(WCHAR* szClassName, WCHAR* szSearch)
```
Comment (MarioHewardt) [inline profiler/src/ProcDumpProfiler.cpp:19]: Make this part of the profiler class

File: src/ProfilerHelpers.c [hunk 3]
```diff
@@ -297,9 +297,10 @@ char* GetEncodedExceptionFilter(char* exceptionFilterCmdLine, unsigned int numDu
     char* exceptionFilter = NULL;
     char* exceptionFilterCur = NULL;
     char tmp[10];
+    size_t len;
```
Comment (MarioHewardt) [inline src/ProfilerHelpers.c:300]: Initialize to 0

File: profiler/inc/profilerstring.h (hunk 1 above)
Comment (oldzhu) [reply profiler/inc/profilerstring.h:364]: Thanks Mario! I will modify the other two and commit, but I have question for this allocation checking with the below two thoughts:
1. There is no checking for other new allocations in the same file.
2. The post below says seems we only need to check new allocation with no-throw
https://stackoverflow.com/questions/5197732/should-one-always-check-if-the-new-operator-worked

**Review by MarioHewardt — APPROVED** [review]

**Pull request merged**